### PR TITLE
Cleaning up the codebase and additional changes in Python SRT generator.

### PR DESCRIPTION
**Pull request by Diptanshu8**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [*] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [*] I have checked that another pull request for this purpose does not exist.
- [*] I have considered, and confirmed that this submission will be valuable to others.
- [*] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [*] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used CCExtractor.
- [ ] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [*] I am an active contributor to CCExtractor.

---
This is majorly a Cleanup PR for cleaning up the codebase. 
A feature to compare the font<->text grids and color<->text grids in Python for generating the output srt file has been added.

### CODE REVIEW

File: api/python_srt_generator.py [hunk 1]
```diff
@@ -43,10 +21,59 @@ def generate_output_srt_time( fh, data):
...
+};
+def comparing_text_font_grids(text, font, color):
+    original_text = text
+    temp_color = []
+    for letter,color_line in zip(original_text,color):
+        color = 0
+        color_flag = 1
+        prev = color_line[0]
+        buff = color_text_start[str(prev)]
+        if "                                " not in letter:
```
Comment (cfsmp3) [inline api/python_srt_generator.py]: Please use a "decent" way to check if a string is empty or not :-)

**Comment by canihavesomecoffee** [comment]
autoconf build is broken:
```
../src/lib_ccx/ccextractor-ccx_encoders_common.o: In function `encode_sub':
/home/bot/ccextractor/linux/../src/lib_ccx/ccx_encoders_common.c:1134: undefined reference to `pass_cc_buffer_to_python'
```

**Comment by Diptanshu8** [comment]
@cfsmp3 Made changes as you suggested. 
@canihavesomecoffee Made additions to Makefile.am.

File: api/extractors/extractor.h [hunk 2]
```diff
@@ -1,10 +1,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include "ccextractor.h"
+#include "../../src/lib_ccx/ccx_common_common.h"
```
Comment (cfsmp3) [inline api/extractors/extractor.h]: You don't need to specify the path, that's what's the include dirs are for.  "ccx_common_common.h" is enough.

**Pull request merged**